components/input/client_test.go

```diff
@@ -267,3 +267,72 @@ func TestClient(t *testing.T) {
 		test.That(t, conn.Close(), test.ShouldBeNil)
 	})
 }
+
+func TestClientRace(t *testing.T) {
+	logger := logging.NewTestLogger(t)
+	listener1, err := net.Listen("tcp", "localhost:0")
```

Comment:
we don't - 0 means it gets auto-assigned. since we pass the listener in directly and don't relinquish control of the port, we won't run into issues of a race where multiple tests will try to use the same port

Reply:
I actually thought the issue was that calls like `net.Listen("localhost:0")` from concurrent/parallel goroutines could very occasionally start listening on the same port. What do you mean by "relinquish control of the port"?

Reply:
the original issue as I understood it was that `TryReserveRandomPort` checks for a port that is free by trying to listen on it, and then stopping the listener and returning the port. The issue then is that concurrent tests might be doing the same thing. If the listener is not stopped, we shouldn't run into the same class of problems

Reply:
Gotcha; that makes more sense. The case I mentioned in my last comment seems uncommonly rare.